Update OneLogin PHP SAML library to 3.0.0.

gonzomir opened the pull request:
Update references to the library to use namespaces. This should fix #17 and allow the plugin to work on hosts with php 7+.

mikeselander commented on inc/admin/namespace.php:
```diff
@@ -4,7 +4,6 @@
 
 use function HumanMade\SimpleSaml\instance;
 use function HumanMade\SimpleSaml\is_sso_enabled_network_wide;
-use OneLogin_Saml2_IdPMetadataParser;
```
This is the opposite of what we should be doing with namespaces. Why take away the `use` statement?

gonzomir replied:
I found having the namespace in the references more readable, but that's maybe just me.

rmccue replied:
Per [coding standards](https://new-engineering.hmn.md/standards/style/php/#namespace-and-class-naming):

> Classes used in a namespaced file should have a `use` statement at the top of the file to allow a quick overview of what is used, and to avoid long absolute references to the classes throughout the file. If you have a lot of functionality from given namespace, you can `use` the entire namespace to avoid excessive use statements.

All the classes here used within the file should be `use`'d at the start rather than absolute refs throughout.

spacedmonkey commented:
@gonzomir Was the composer update run with this flag `--optimize-autoloader` ? If you are going to check the vendor directory, might be worth doing this.

gonzomir commented:
@spacedmonkey No, I didn't use `--optimize-autoloader` initially. Just updated the PR with optimised autoloader files and also cleaned up from development dependencies.

Since OneLogin PHP SAML 3.0 now doesn't include xmlseclibs, should we update the README mentioning one should `composer install`? @mikeselander Any thoughts on this?

spacedmonkey commented:
When testing this I got the following error. 
![skyscanner local_sso_login__redirect_to http 3a 2f 2fskyscanner local 2fwp-admin 2f](https://user-images.githubusercontent.com/237508/48206001-45f9c900-e365-11e8-8042-7537011a422d.png)

This is because I installed the plugin via composer. It has issues because of the hard coded autoloader and vendor directories in the project. 

Here is a workaround / fix for this issue and it comes from the S3-Uploads plugin via @rmccue . See https://github.com/humanmade/S3-Uploads/commit/da4bdd7775842b622232da65c7bc0bc1b9913f53

Change the following line in plugin.php 
```
require_once __DIR__ . '/vendor/autoload.php';
```
to 
```
if ( ! class_exists('\\OneLogin\\Saml2\\Auth') ) {
    require_once __DIR__ . '/vendor/autoload.php';
}
```

shadyvb commented:
Tagged v0.2 and updated changelog.

shadyvb commented:
FYI the git ignore rules was updated to allow for the new dependency to be committed. Fixed and tagged as 0.2.1.
